src/dora/kaggle.py

```
@@ -0,0 +1,78 @@
+        """
+        Download a Kaggle dataset from kagglehub and return the path to the downloaded file.
+
+        :param dataset_id: The 'owner/dataset-name' identifier of the dataset to download.
+        :return: The path to the downloaded file.
+        """
+        logging.info(f"Downloading dataset {dataset_id}")
+        try:
+            dataset_path = kagglehub.dataset_download(dataset_id)
+            dataset_download_directory = Path(dataset_path)
+        except RuntimeError as e:
+            raise ValueError(f"Failed to download dataset: {e}")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)